pkg/storage/action_cache_asset_store.go

```diff
@@ -204,22 +156,10 @@ func (rs *actionCacheAssetStore) Put(ctx context.Context, ref *asset.AssetRefere
 	}
 	var action *remoteexecution.Action
 	var command *remoteexecution.Command
-	if commandGenerator, err := qualifier.QualifiersToCommand(ref.Qualifiers); err != nil || len(ref.Uris) > 1 {
-		// Create the action with the qualifier directory as the input root
-		action, command, err = assetReferenceToAction(ref, directoryDigest)
-		if err != nil {
-			return err
-		}
-	} else {
-		command = commandGenerator(ref.Uris[0])
-		commandDigest, err := ProtoToDigest(command)
-		if err != nil {
-			return err
-		}
-		action = &remoteexecution.Action{
-			CommandDigest:   commandDigest,
-			InputRootDigest: EmptyDigest,
-		}
+	// Create the action with the qualifier directory as the input root
+	action, command, err = assetReferenceToAction(ref, directoryDigest)
+	if err != nil {
+		return err
```

Comment:
This change makes assets with the special qualifiers we check for in `QualifiersToCommand` to unfetchable from the asset server.

Either we need to revert this bit or apply the corresponding change (i.e dropping `QualifiersToCommand`) from `Get`.